src/vos/vos_dtx.c

```diff
@@ -1568,30 +1581,6 @@ vos_dtx_prepared(struct dtx_handle *dth)
 	D_ASSERT(cont != NULL);
 
 	dtx = umem_off2ptr(&cont->vc_pool->vp_umm, dth->dth_ent);
-	if (dth->dth_intent == DAOS_INTENT_UPDATE) {
-		d_iov_t	kiov;
-		d_iov_t	riov;
-
-		/* There is CPU yield during the bulk transfer, then it is
-		 * possible that some others (rebuild) abort this DTX by race.
-		 * So we need to locate (or verify) DTX via its ID instead of
-		 * directly using the dth_ent.
```

Comment:
this case would not happen any more?

Reply:
Yes. Originally, we create the object (if non-exist) when vos_update_begin, then bulk_transfer, and then vos_update_end. The DTX entry is generated when the object is created, and the DTX prepare is called during vos_update_end, so there may CPU yield between DTX entry generated and DTX prepared. But now, we create object (if non-exist) in vos_update_end. So there is CPU yield between DTX entry generated and DTX prepared.